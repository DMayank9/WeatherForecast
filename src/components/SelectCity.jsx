import React, { useState } from 'react';
import { Row, Col, FormControl, Button } from 'react-bootstrap';

// Provide a field to enter city for which weather forecast will be done
const SelectCity = ({ fetchWeatherData }) => {
  const [city, setCity] = useState('');
  return (
    <>
      <Row >
        <Col xs={3} md={3} lg={3} className="text-center">
          <FormControl placeholder="Enter the City name"
            onChange={(event) => setCity(event.target.value)}
            value={city}
          >
          </FormControl>
        </Col>
        <Col xs={3} md={3} lg={3}>
          <Button onClick={() => fetchWeatherData(city)}>Forecast Weather</Button>
        </Col>
      </Row>
    </>
  )
}

export default SelectCity;
