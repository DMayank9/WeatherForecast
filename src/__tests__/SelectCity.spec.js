import { mount } from "enzyme";
import SelectCity from "../components/SelectCity";
describe("Test select input element", () => {
  let fullDOMRenderWrapper;
  beforeEach(() => {
    fullDOMRenderWrapper = mount(<SelectCity />);
  });
  it("has two elements", () => {
    expect(fullDOMRenderWrapper.find("input").length).toEqual(1);
  });
  it("should allow to type in the enter city input field", () => {
    fullDOMRenderWrapper.find("input").simulate("change", {
      target: { value: "Delhi" },
    });
    fullDOMRenderWrapper.update();
    expect(
      fullDOMRenderWrapper.find("input").prop("value")
    ).toEqual("Delhi");
  });
  it("has a single button", () => {
    expect(fullDOMRenderWrapper.find("button").length).toEqual(1);
  });
  it("Renders Weather App", () => {
    expect(fullDOMRenderWrapper).toMatchSnapshot();
  });
});
