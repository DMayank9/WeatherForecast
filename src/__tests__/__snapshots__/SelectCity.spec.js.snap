// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test select input element Renders Weather App 1`] = `ReactWrapper {}`;
